slidez-fe-core/src/pages/dashboard/UserProfile.tsx

```diff
@@ -1,38 +1,125 @@
+    firstName: string | undefined
+    lastName: string | undefined
+    email: string | undefined
+}
+
+const initialValuesUserData: User = {
+    id: '',
+    firstName: '',
+    lastName: '',
+    email: '',
+}
+
+export interface Password {
+    id: string | undefined
+    password: string
+    confirmPassword: string
+}
+
+const initialValuesPassword: Password = {
+    id: '',
+    password: '',
+    confirmPassword: '',
+}
 
-const UserProfile = () => {
+const UserProfile = ({
+    onUpdateUserProfile,
+    onUpdatePassword,
+}: UpdateProps) => {
     const JWT = 'jwt'
     const refreshJWT = 'refresh_jwt'
     const dispatch = useAppDispatch()
     const dropdownRef = useRef<HTMLInputElement>(null)
     const [token, setToken] = useState('')
-    const [userFirstName, setUserFirstName] = useState<string | undefined>('')
-    const [isActive, setIsActive] = useDetectOutsideClick(dropdownRef, false)
-    const [userLastName, setUserLastName] = useState<string | undefined>('')
-    const [userEmail, setUserEmail] = useState<string | undefined>('')
+    const [isActive, setIsActive] = useState(false)
     const [logo, setLogo] = useState('')
+    const [openEditProfile, setOpenEditProfile] = useState(false)
+    const [isPasswordRevealed, setIsPasswordRevealed] = React.useState(false)
+    const [isConfirmPasswordRevealed, setIsConfirmPasswordRevealed] =
+        React.useState(false)
+    const [viewErrors, setViewErrors] = React.useState(false)
+    const [userData, setUserData] = useState<User>(initialValuesUserData)
+    const [userPassword, setUserPassword] = useState<Password>(
+        initialValuesPassword
+    )
+    const [isUpdatedUserData, setIsUpdatedUserData] = useState(false)
 
     useEffect(() => {
-        if (token.length > 0) {
+        if (token.length > 0 && !isUpdatedUserData) {
             performLoginByToken()
-            handleLogo()
+            setIsUpdatedUserData(true)
             return
         }
         getAccessToken()
+        handleLogo()
     })
 
+    const onOpenEditDialog = () => {
+        setOpenEditProfile(true)
+        handleDropDown()
+    }
+
+    const onCloseEditDialog = () => {
+        setOpenEditProfile(false)
+    }
+
+    const onRevealPasswordClick = () => {
+        setIsPasswordRevealed(!isPasswordRevealed)
+        revealPassword('update-password-input')
+    }
+
+    const onRevealConfirmPasswordClick = () => {
+        setIsConfirmPasswordRevealed(!isConfirmPasswordRevealed)
+        revealPassword('update-confirmPassword-input')
+    }
+
+    const handleUpdateUserProfile = async (data: any) => {
+        onUpdateUserProfile(data)
+        setIsUpdatedUserData(false)
+    }
+
+    const handleUpdatePassword = async (
+        id: string | undefined,
+        password: string,
+        confirmPassword: string
+    ) => {
+        const pass: Password = {
+            id: id,
+            password: password,
+            confirmPassword: confirmPassword,
```

Comment:
Check password and confirmPassword on the frontend, then send the only password to the backend, id => userId

Reply:
I added validation this fields to cannot send only field 'password' without 'confirm password'